### ISSUE TITLE
[0.9.5-nightly-66a31ee] tsm1 compaction... not compacting?

### ISSUE THREAD

**Issue by dswarbrick** [influxdb/influxdb#4699]
Apologies in advance if this is intended behaviour. I've noticed a massive change in InfluxDB tsm1's disk utilization patterns. We were running 0.9.5-nightly-66a31ee with quite satisfactory results on a test server, using about 11 GB of disk space per day, and deployed a second server (not clustered) with 0.9.5-nightly-780df57.

We immediately noticed that the latter version was using a lot more disk space, despite being fed with the same metrics. It used disk space at an alarming rate, using over 65 GB in less than 24 hours. I dropped the DB to start afresh, but it again looked like it was climbing fast. I upgraded to 0.9.5-nightly-ef84e33, and noticed a large drop in disk space used, as if something had been compacted. However the trend was still the same shortly afterwards.

What I've found is that each time I restart InfluxDB, it seems to perform a compaction, and then resumes it's former rapid consumption of space. This is completely different to how it behaved in 66a31ee, which is why I suspect something has changed or been broken with respect to compaction.

I've attached an annotated graph of disk space utilization.

![influxdb-tsm1](https://cloud.githubusercontent.com/assets/1667905/10997009/f16cb4f2-8489-11e5-924f-29ccde26bef1.png)

**Comment by dswarbrick** [comment]
At a rough guess, I suspect this 1233383 may have introduced the behaviour change.

**Comment by otoolep** [comment]
Thanks @dswarbrick -- we'll take a look.

**Comment by otoolep** [comment]
@dswarbrick  - can you tell about the source of your metrics? How many senders, using which input? I think you use HTTP, right?

**Comment by dswarbrick** [comment]
@otoolep That's right, we're using an inhouse-developed InfluxDB (http) output plugin for collectd. We have about 200 hosts currently shooting data at it.

**Comment by otoolep** [comment]
We running tests here of the latest nightly and the tsm1 engine. We are writing 100B points, over HTTP/ and will report back.

**Comment by otoolep** [comment]
@mjdesa -- can you post details of the test here?

**Comment by mjdesa** [comment]
- 200k series
- 1M points per series
- 4 tags
- 1 field
- 10s interval between timestamps
- 0s between writes
- 10 concurrent writers

**Comment by JulienChampseix** [comment]
@mjdesa, what's your config file during this test ?

**Comment by mjdesa** [comment]
```
channel_buffer_size = 15000

[write]
concurrency = 10
batch_size = 10000
batch_interval = "0s"
database = "stress"
precision = "s"
address = "192.168.1.247:8086"
reset_database = true
start_date = "2006-Jan-02"

[[series]]
tick = "10s"
jitter = true
point_count = 1000000 # number of points that will be written for each of the series
measurement = "cpu"
series_count = 200000

[[series.tag]]
key = "host"
value = "server"

[[series.tag]]
key = "location"
value = "loc"

[[series.tag]]
key = "tag"
value = "server"

[[series.tag]]
key = "othertag"
value = "server"

[[series.field]]
key = "value"
type = "float64"
```

**Comment by otoolep** [comment]
Which config? InfluxDB or the stress tool?

**Comment by mjdesa** [comment]
```
### Welcome to the InfluxDB configuration file.

# Once every 24 hours InfluxDB will report anonymous data to m.influxdb.com
# The data includes raft id (random 8 bytes), os, arch, version, and metadata.
# We don't track ip addresses of servers reporting. This is only used
# to track the number of instances running and the versions, which
# is very helpful for us.
# Change this option to true to disable reporting.
reporting-disabled = true

###
### [meta]
###
### Controls the parameters for the Raft consensus group that stores metadata
### about the InfluxDB cluster.
###

[meta]
  dir = "/var/opt/influxdb/meta"
  hostname = "localhost"
  bind-address = ":8088"
  retention-autocreate = true
  election-timeout = "1s"
  heartbeat-timeout = "1s"
  leader-lease-timeout = "500ms"
  commit-timeout = "50ms"
  cluster-tracing = false

###
### [data]
###
### Controls where the actual shard data for InfluxDB lives.
###

[data]
  dir = "/var/opt/influxdb/data"

  engine = "tsm1"

  # The following WAL settings are for the b1 storage engine used in 0.9.2. They won't
  # apply to any new shards created after upgrading to a version > 0.9.3.
  max-wal-size = 104857600
  wal-flush-interval = "10m0s"
  wal-partition-flush-delay = "2s"

  # These are the WAL settings for the storage engine >= 0.9.3
  wal-dir = "/var/opt/influxdb/wal"
  wal-logging-enabled = true

  # When a series in the WAL in-memory cache reaches this size in bytes it is marked as ready to
  # flush to the index
  # wal-ready-series-size = 30720

  # Flush and compact a partition once this ratio of series are over the ready size
  # wal-compaction-threshold = 0.5

  # Force a flush and compaction if any series in a partition gets above this size in bytes
  # wal-max-series-size = 1048576

  # Force a flush of all series and full compaction if there have been no writes in this
  # amount of time. This is useful for ensuring that shards that are cold for writes don't
  # keep a bunch of data cached in memory and in the WAL.
  # wal-flush-cold-interval = "5s"

  # Force a partition to flush its largest series if it reaches this approximate size in
  # bytes. Remember there are 5 partitions so you'll need at least 5x this amount of memory.
  # The more memory you have, the bigger this can be.
  # wal-partition-size-threshold = 20971520

###
### [cluster]
###
### Controls non-Raft cluster behavior, which generally includes how data is
### shared across shards.
###

[cluster]
  shard-writer-timeout = "5s"
  force-remote-mapping = false
  write-timeout = "5s"
  shard-mapper-timeout = "5s"

###
### [retention]
###
### Controls the enforcement of retention policies for evicting old data.
###

[retention]
  enabled = true
  check-interval = "30m0s"

[shard-precreation]
  enabled = true
  check-interval = "10m0s"
  advance-period = "30m0s"

###
### [admin]
###
### Controls the availability of the built-in, web-based admin interface.
###

[admin]
  enabled = true
  bind-address = ":8083"
  https-enabled = false
  https-certificate = "/etc/ssl/influxdb.pem"

[monitor]
  store-enabled = true
  store-database = "_internal"
  store-interval = "10s"

###
### [http]
###
### Controls how the HTTP endpoints are configured. These are the primary
### mechanism for getting data into and out of InfluxDB.
###

[http]
  enabled = true
  bind-address = ":8086"
  auth-enabled = false
  log-enabled = false
  write-tracing = false
  pprof-enabled = false
  https-enabled = false
  https-certificate = "/etc/ssl/influxdb.pem"

###
### [[graphite]]
###
### Controls one or many listeners for Graphite data.
###

[[graphite]]
  enabled = false
  # bind-address = ":2003"
  # protocol = "tcp"
  # consistency-level = "one"
  # name-separator = "."
  # name-position = "last"

  # These next lines control how batching works. You should have this enabled
  # otherwise you could get dropped metrics or poor performance. Batching
  # will buffer points in memory if you have many coming in.

  # batch-size = 1000 # will flush if this many points get buffered
  # batch-pending = 5 # number of batches that may be pending in memory
  # batch-timeout = "1s" # will flush at least this often even if we haven't hit buffer limit

  ## "name-schema" configures tag names for parsing the metric name from graphite protocol;
  ## separated by `name-separator`.
  ## The "measurement" tag is special and the corresponding field will become
  ## the name of the metric.
  ## e.g. "type.host.measurement.device" will parse "server.localhost.cpu.cpu0" as
  ## {
  ##     measurement: "cpu",
  ##     tags: {
  ##         "type": "server",
  ##         "host": "localhost,
  ##         "device": "cpu0"
  ##     }
  ## }
  # name-schema = "type.host.measurement.device"

  ## If set to true, when the input metric name has more fields than `name-schema` specified,
  ## the extra fields will be ignored.
  ## Otherwise an error will be logged and the metric rejected.
  # ignore-unnamed = true

###
### [collectd]
###
### Controls the listener for collectd data.
###

[collectd]
  enabled = false
  bind-address = ":25826"
  database = "collectd"
  retention-policy = ""
  batch-size = 1000
  batch-pending = 5
  batch-timeout = "10s"
  typesdb = "/usr/share/collectd/types.db"

###
### [opentsdb]
###
### Controls the listener for OpenTSDB data.
###

[opentsdb]
  enabled = false
  # bind-address = ":4242"
  # database = "opentsdb"
  # retention-policy = ""
  # consistency-level = "one"
  # tls-enabled = false
  # certificate = "/etc/ssl/influxdb.pem"
  # batch-size = 1000
  # batch-pending = 5
  # batch-timeout = "1s"

###
### [udp]
###
### Controls the listener for InfluxDB line protocol data via UDP.
###

# TODO: Including the udp table prevents service from starting (even if disabled)
# [udp]
#   enabled = false
  # bind-address = ""
  # database = ""
  # batch-size = 0
  # batch-timeout = "0"

###
### [monitoring]
###

[monitoring]
  enabled = true
  write-interval = "24h"

###
### [continuous_queries]
###
### Controls how continuous queries are run within InfluxDB.
###

[continuous_queries]
  log-enabled = true
  enabled = true
  recompute-previous-n = 2
  recompute-no-older-than = "10m"
  compute-runs-per-interval = 10
  compute-no-more-than = "2m"

###
### [hinted-handoff]
###
### Controls the hinted handoff feature, which allows nodes to temporarily
### store queued data when one node of a cluster is down for a short period
### of time.
###

[hinted-handoff]
  enabled = true
  dir = "/var/opt/influxdb/hh"
  max-size = 1073741824
  max-age = "168h"
  retry-rate-limit = 0
  retry-interval = "1s"
```

**Comment by JulienChampseix** [comment]
@mjdesa, thanks ! So that's the config used during the stress test and dataset you are using, i'm right? Which tools are you using with this stress dataset ?

**Comment by mjdesa** [comment]
Yup.

The [tool](https://github.com/influxdb/influxdb/tree/master/cmd/influx_stress) I'm using to stress the database is called `influx_stress`.

**Comment by JulienChampseix** [comment]
What's the best metrics into _internal to follow the bench stress in termnof queries ? 
Why i'm resquesting that? I migrated from 0.9.4.2 to the nightly 0.9.5 from wednesday and the startup was slow due WAL writes to index and having feeling space was increasing fast using tsm1. I would by this way benchmark again my server for avoid any confusion.

**Comment by mjdesa** [comment]
Depends largely on what you're looking to benchmark. 

This query will tell you the write throughput
```sql
select derivative(pointReq) from "write"
```

Typically we monitor resource usage for a machine externally with Telegraf or Collectd.

**Comment by otoolep** [comment]
Here are the graphs so far.
![g](https://cloud.githubusercontent.com/assets/536312/11009158/e40daf32-8489-11e5-8363-c5e0ed975bdf.png)

**Comment by otoolep** [comment]
Not sure, but compaction does look like it may have ceased on system too -- disk usage and RSS going up steadily.

**Comment by otoolep** [comment]
Yeah, ton of tsm files in 4.

```
root@sf7:/var/opt/influxdb/data/stress/default/4# ls
0000845.tsm1  0000920.tsm1  0000996.tsm1  0001071.tsm1  0001145.tsm1  0001221.tsm1  0001297.tsm1  0001372.tsm1  0001446.tsm1  0001522.tsm1  0001598.tsm1  0001674.tsm1  0001748.tsm1  0001824.tsm1  0001900.tsm1  0001971.tsm1.check
0000847.tsm1  0000922.tsm1  0000998.tsm1  0001073.tsm1  0001147.tsm1  0001223.tsm1  0001299.tsm1  0001374.tsm1  0001448.tsm1  0001524.tsm1  0001600.tsm1  0001676.tsm1  0001750.tsm1  0001826.tsm1  0001901.tsm1  0001972.tsm1
0000849.tsm1  0000924.tsm1  0001000.tsm1  0001075.tsm1  0001149.tsm1  0001225.tsm1  0001301.tsm1  0001376.tsm1  0001450.tsm1  0001526.tsm1  0001602.tsm1  0001678.tsm1  0001752.tsm1  0001828.tsm1  0001904.tsm1  0001972.tsm1.check
0000851.tsm1  0000926.tsm1  0001002.tsm1  0001077.tsm1  0001151.tsm1  0001227.tsm1  0001303.tsm1  0001378.tsm1  0001452.tsm1  0001528.tsm1  0001604.tsm1  0001680.tsm1  0001754.tsm1  0001830.tsm1  0001906.tsm1  _22046.wal
0000853.tsm1  0000928.tsm1  0001004.tsm1  0001079.tsm1  0001153.tsm1  0001229.tsm1  0001304.tsm1  0001380.tsm1  0001454.tsm1  0001530.tsm1  0001606.tsm1  0001681.tsm1  0001756.tsm1  0001832.tsm1  0001908.tsm1  _22047.wal
0000854.tsm1  0000930.tsm1  0001005.tsm1  0001081.tsm1  0001155.tsm1  0001231.tsm1  0001307.tsm1  0001382.tsm1  0001456.tsm1  0001532.tsm1  0001608.tsm1  0001682.tsm1  0001758.tsm1  0001834.tsm1  0001910.tsm1  _22048.wal
0000856.tsm1  0000932.tsm1  0001007.tsm1  0001083.tsm1  0001157.tsm1  0001233.tsm1  0001309.tsm1  0001384.tsm1  0001458.tsm1  0001534.tsm1  0001610.tsm1  0001685.tsm1  0001760.tsm1  0001836.tsm1  0001912.tsm1  _22049.wal
0000858.tsm1  0000934.tsm1  0001009.tsm1  0001085.tsm1  0001159.tsm1  0001235.tsm1  0001311.tsm1  0001386.tsm1  0001460.tsm1  0001536.tsm1  0001612.tsm1  0001687.tsm1  0001762.tsm1  0001838.tsm1  0001914.tsm1  _22050.wal
0000860.tsm1  0000936.tsm1  0001011.tsm1  0001087.tsm1  0001161.tsm1  0001237.tsm1  0001313.tsm1  0001388.tsm1  0001462.tsm1  0001538.tsm1  0001614.tsm1  0001689.tsm1  0001764.tsm1  0001840.tsm1  0001916.tsm1  _22051.wal
0000862.tsm1  0000938.tsm1  0001013.tsm1  0001089.tsm1  0001163.tsm1  0001239.tsm1  0001315.tsm1  0001390.tsm1  0001464.tsm1  0001540.tsm1  0001616.tsm1  0001691.tsm1  0001766.tsm1  0001842.tsm1  0001918.tsm1  _22052.wal
0000864.tsm1  0000940.tsm1  0001015.tsm1  0001091.tsm1  0001165.tsm1  0001241.tsm1  0001317.tsm1  0001392.tsm1  0001466.tsm1  0001542.tsm1  0001618.tsm1  0001693.tsm1  0001768.tsm1  0001844.tsm1  0001920.tsm1  _22053.wal
0000866.tsm1  0000942.tsm1  0001017.tsm1  0001093.tsm1  0001167.tsm1  0001243.tsm1  0001319.tsm1  0001394.tsm1  0001468.tsm1  0001544.tsm1  0001620.tsm1  0001695.tsm1  0001770.tsm1  0001846.tsm1  0001922.tsm1  _22054.wal
0000868.tsm1  0000944.tsm1  0001019.tsm1  0001095.tsm1  0001169.tsm1  0001245.tsm1  0001321.tsm1  0001396.tsm1  0001470.tsm1  0001546.tsm1  0001622.tsm1  0001697.tsm1  0001772.tsm1  0001848.tsm1  0001924.tsm1  _22055.wal
0000870.tsm1  0000946.tsm1  0001021.tsm1  0001097.tsm1  0001171.tsm1  0001247.tsm1  0001323.tsm1  0001398.tsm1  0001472.tsm1  0001548.tsm1  0001624.tsm1  0001699.tsm1  0001774.tsm1  0001850.tsm1  0001926.tsm1  _22056.wal
0000872.tsm1  0000948.tsm1  0001023.tsm1  0001099.tsm1  0001173.tsm1  0001249.tsm1  0001325.tsm1  0001400.tsm1  0001474.tsm1  0001550.tsm1  0001626.tsm1  0001701.tsm1  0001776.tsm1  0001852.tsm1  0001928.tsm1  _22057.wal
0000873.tsm1  0000950.tsm1  0001025.tsm1  0001101.tsm1  0001175.tsm1  0001251.tsm1  0001327.tsm1  0001402.tsm1  0001476.tsm1  0001552.tsm1  0001628.tsm1  0001703.tsm1  0001778.tsm1  0001854.tsm1  0001930.tsm1  _22058.wal
0000876.tsm1  0000952.tsm1  0001027.tsm1  0001103.tsm1  0001177.tsm1  0001253.tsm1  0001329.tsm1  0001404.tsm1  0001478.tsm1  0001554.tsm1  0001630.tsm1  0001705.tsm1  0001780.tsm1  0001856.tsm1  0001932.tsm1  _22059.wal
0000878.tsm1  0000954.tsm1  0001029.tsm1  0001104.tsm1  0001179.tsm1  0001254.tsm1  0001331.tsm1  0001406.tsm1  0001480.tsm1  0001556.tsm1  0001632.tsm1  0001707.tsm1  0001782.tsm1  0001858.tsm1  0001934.tsm1  _22060.wal
0000880.tsm1  0000956.tsm1  0001031.tsm1  0001105.tsm1  0001181.tsm1  0001257.tsm1  0001333.tsm1  0001408.tsm1  0001482.tsm1  0001558.tsm1  0001634.tsm1  0001709.tsm1  0001784.tsm1  0001860.tsm1  0001936.tsm1  _22061.wal
0000882.tsm1  0000958.tsm1  0001033.tsm1  0001107.tsm1  0001183.tsm1  0001259.tsm1  0001335.tsm1  0001410.tsm1  0001484.tsm1  0001560.tsm1  0001636.tsm1  0001711.tsm1  0001786.tsm1  0001862.tsm1  0001938.tsm1  _22062.wal
0000884.tsm1  0000960.tsm1  0001035.tsm1  0001108.tsm1  0001185.tsm1  0001261.tsm1  0001337.tsm1  0001412.tsm1  0001486.tsm1  0001562.tsm1  0001638.tsm1  0001713.tsm1  0001788.tsm1  0001864.tsm1  0001940.tsm1  _22063.wal
0000886.tsm1  0000962.tsm1  0001037.tsm1  0001111.tsm1  0001187.tsm1  0001263.tsm1  0001339.tsm1  0001414.tsm1  0001488.tsm1  0001564.tsm1  0001640.tsm1  0001715.tsm1  0001790.tsm1  0001866.tsm1  0001942.tsm1  _22064.wal
0000888.tsm1  0000964.tsm1  0001039.tsm1  0001113.tsm1  0001189.tsm1  0001265.tsm1  0001341.tsm1  0001416.tsm1  0001490.tsm1  0001566.tsm1  0001642.tsm1  0001717.tsm1  0001792.tsm1  0001868.tsm1  0001944.tsm1  _22065.wal
0000890.tsm1  0000966.tsm1  0001041.tsm1  0001115.tsm1  0001191.tsm1  0001267.tsm1  0001343.tsm1  0001418.tsm1  0001492.tsm1  0001568.tsm1  0001644.tsm1  0001719.tsm1  0001794.tsm1  0001870.tsm1  0001946.tsm1  _22066.wal
0000892.tsm1  0000968.tsm1  0001043.tsm1  0001117.tsm1  0001193.tsm1  0001269.tsm1  0001345.tsm1  0001419.tsm1  0001494.tsm1  0001570.tsm1  0001646.tsm1  0001721.tsm1  0001796.tsm1  0001872.tsm1  0001948.tsm1  _22067.wal
0000894.tsm1  0000970.tsm1  0001045.tsm1  0001119.tsm1  0001195.tsm1  0001271.tsm1  0001347.tsm1  0001420.tsm1  0001496.tsm1  0001572.tsm1  0001648.tsm1  0001723.tsm1  0001798.tsm1  0001874.tsm1  0001950.tsm1  _22068.wal
0000896.tsm1  0000972.tsm1  0001047.tsm1  0001121.tsm1  0001197.tsm1  0001273.tsm1  0001349.tsm1  0001423.tsm1  0001498.tsm1  0001574.tsm1  0001650.tsm1  0001725.tsm1  0001800.tsm1  0001876.tsm1  0001952.tsm1  _22069.wal
0000898.tsm1  0000974.tsm1  0001049.tsm1  0001123.tsm1  0001199.tsm1  0001275.tsm1  0001351.tsm1  0001425.tsm1  0001500.tsm1  0001576.tsm1  0001652.tsm1  0001727.tsm1  0001802.tsm1  0001878.tsm1  0001954.tsm1  _22070.wal
0000900.tsm1  0000976.tsm1  0001051.tsm1  0001125.tsm1  0001201.tsm1  0001277.tsm1  0001353.tsm1  0001427.tsm1  0001502.tsm1  0001578.tsm1  0001654.tsm1  0001729.tsm1  0001804.tsm1  0001880.tsm1  0001956.tsm1  _22071.wal
0000902.tsm1  0000978.tsm1  0001053.tsm1  0001127.tsm1  0001203.tsm1  0001279.tsm1  0001355.tsm1  0001429.tsm1  0001504.tsm1  0001580.tsm1  0001656.tsm1  0001731.tsm1  0001806.tsm1  0001882.tsm1  0001958.tsm1  _22072.wal
0000904.tsm1  0000980.tsm1  0001055.tsm1  0001129.tsm1  0001205.tsm1  0001281.tsm1  0001356.tsm1  0001431.tsm1  0001506.tsm1  0001582.tsm1  0001658.tsm1  0001733.tsm1  0001808.tsm1  0001884.tsm1  0001960.tsm1  _22073.wal
0000906.tsm1  0000982.tsm1  0001057.tsm1  0001131.tsm1  0001207.tsm1  0001283.tsm1  0001357.tsm1  0001433.tsm1  0001508.tsm1  0001584.tsm1  0001660.tsm1  0001734.tsm1  0001810.tsm1  0001886.tsm1  0001962.tsm1  fields
0000908.tsm1  0000984.tsm1  0001059.tsm1  0001133.tsm1  0001209.tsm1  0001285.tsm1  0001360.tsm1  0001435.tsm1  0001510.tsm1  0001586.tsm1  0001662.tsm1  0001735.tsm1  0001812.tsm1  0001888.tsm1  0001964.tsm1  ids
0000910.tsm1  0000985.tsm1  0001061.tsm1  0001135.tsm1  0001211.tsm1  0001286.tsm1  0001362.tsm1  0001437.tsm1  0001512.tsm1  0001588.tsm1  0001664.tsm1  0001738.tsm1  0001814.tsm1  0001890.tsm1  0001966.tsm1  series
0000912.tsm1  0000988.tsm1  0001063.tsm1  0001137.tsm1  0001213.tsm1  0001289.tsm1  0001364.tsm1  0001439.tsm1  0001514.tsm1  0001590.tsm1  0001666.tsm1  0001740.tsm1  0001816.tsm1  0001892.tsm1  0001968.tsm1
0000914.tsm1  0000990.tsm1  0001065.tsm1  0001139.tsm1  0001215.tsm1  0001291.tsm1  0001366.tsm1  0001440.tsm1  0001516.tsm1  0001592.tsm1  0001668.tsm1  0001742.tsm1  0001818.tsm1  0001894.tsm1  0001969.tsm1
0000916.tsm1  0000992.tsm1  0001067.tsm1  0001141.tsm1  0001217.tsm1  0001293.tsm1  0001368.tsm1  0001441.tsm1  0001518.tsm1  0001594.tsm1  0001670.tsm1  0001744.tsm1  0001820.tsm1  0001896.tsm1  0001970.tsm1
0000918.tsm1  0000994.tsm1  0001069.tsm1  0001143.tsm1  0001219.tsm1  0001295.tsm1  0001370.tsm1  0001444.tsm1  0001520.tsm1  0001596.tsm1  0001672.tsm1  0001746.tsm1  0001822.tsm1  0001898.tsm1  0001971.tsm1
root@sf7:/var/opt/influxdb/data/stress/default/4# 
```

**Comment by otoolep** [comment]
The other 2 directories has about ~10 files. I just bounced the test node to see what it does to the files in this directory.

**Comment by otoolep** [comment]
No immediate change to disk usage.

**Comment by otoolep** [comment]
Will check in a few minutes.

**Comment by otoolep** [comment]
Yeah, I had a compaction kick in after the restart.

**Comment by otoolep** [comment]
```
# ls -ahl | wc -l
707
<.....wait a few minutes.....>
# ls -ahl | wc -l
291
```

**Comment by otoolep** [comment]
![g1](https://cloud.githubusercontent.com/assets/536312/11010626/13a1ef82-8495-11e5-9735-78e14d9d1d6b.png)

**Comment by otoolep** [comment]
Look like what you see @dswarbrick ?

**Comment by dswarbrick** [comment]
@otoolep yes, looks very similar to what we're seeing. I'm just trying a restart now too. I have 2092 files before restart.
```
# ls -l /srv/influxdb/data/perfdb/default/2 | wc -l
2092
```

**Comment by mjdesa** [comment]
@otoolep looks like there was a dramatic drop in write ops after restart

**Comment by dswarbrick** [comment]
Half an hour later, the file count hasn't decreased at all. In fact it seems to have just added to what's there. There are now 2161 files in the directory.
![screenshot from 2015-11-07 00-26-58](https://cloud.githubusercontent.com/assets/1667905/11011268/922d0926-84e6-11e5-972b-e2e00694c048.png)
There was a small dip in disk usage, but it's climbing again. Note also the large read IOPS spike for about 10 mins after the restart. During this period, no queries can be serviced (neither read nor write), and a single CPU core is maxed out. After a while, normal operation resumes, and all CPU cores start spreading the load.

**Comment by mjdesa** [comment]
Just got in this morning an it looks like after a while things started to compact.
![lvavhuk9e1rhto0z7rn5m3khuqzhv7-pzsknuqvhjxq](https://cloud.githubusercontent.com/assets/1477539/11043557/27a79fec-86d2-11e5-98b5-76c45bac19fa.png)

**Comment by dswarbrick** [comment]
I nuked my DB again over the weekend, and for a while it looked good, but then started to trend upwards again rapidly. This morning I restarted influxd and saw a massive drop in disk space utilization, from 126 GB down to 14 GB. That's some pretty serious compaction - that wasn't happening automatically. Since this morning, it has climbed to almost 120 GB again, and I _know_ we're not feeding _that_ much data into it.

I would go back to an earlier nightly if I could, because it was actually performing very well for a while.

![screen shot 2015-11-09 at 20 27 55](https://cloud.githubusercontent.com/assets/1667905/11044204/c84d08b2-8720-11e5-986b-3db0164bfb02.png)

**Comment by dswarbrick** [comment]
In addition to the rapid disk consumption, we now also see regular (approx. every 30 minutes) IO read spikes, hitting 2K IOPS (where we have it capped with cgroups), for a few minutes. We never saw anything higher than about 300 IOPS with the 66a31ee nightly, and only for very brief bursts.

**Comment by mjdesa** [comment]
Yeah this is very odd.

In my previous post I didn't mean to imply that things were okay, just that it seemed as if things did actually start to compact at some point.

**Comment by dswarbrick** [comment]
I again started with a fresh DB, running 0.9.5-nightly-905437c, to watch the behaviour. Once again, it seemed to do massive compactions whenever I restarted influxd. I downgraded to 66a31ee (built from source) as an experiment, and initially the rate of disk consumption seemed a bit slower, but starting climbing again rapidly later on. It also resulted in a large compaction when I restarted it. So I'm starting to suspect that the compaction has never been quite right.

![screen shot 2015-11-11 at 12 21 21](https://cloud.githubusercontent.com/assets/1667905/11090149/d39be49a-886f-11e5-93b4-aa53d32f51ad.png)

Note that throughout my testing, the points flowing into InfluxDB have remained relatively constant.

**Comment by dswarbrick** [comment]
Any progress on this? I have again wiped my data directory, and looking at my disk space utilization graphs, I'm starting to get the feeling that it all goes south as soon as there is 2 GB of data in the DB.

This strikes me as more than a little bit of a coincidence, since the commit comment in 1233383 is:
> Ensure compactions don't create files much larger than 2GB.

**Comment by otoolep** [comment]
We are heavily refactoring the tsm1 code as I type and and plan to fix this
as part of that work. We hope have it ready soon.

**Comment by dswarbrick** [comment]
Thanks @otoolep for the update. I look forward to trying the new build as soon as it's ready.

**Comment by pauldix** [comment]
Should be fixed with 0.9.6
